mater/lib/src/stores/filestore.rs

```diff
@@ -90,6 +99,86 @@ where
         }
     }
 
+    let index_offset = position;
+    let single_width_index =
+        SingleWidthIndex::new(Sha256::output_size() as u32, entries.len() as u64, entries);
+    let index = Index::MultihashIndexSorted(MultihashIndexSorted::from_single_width(
+        SHA_256_CODE,
+        single_width_index.into(),
+    ));
+    writer.write_index(&index).await?;
+
+    writer.get_inner_mut().rewind().await?;
+    let header = CarV2Header::new(
+        false,
+        car_v1_start.try_into().unwrap(),
+        (index_offset - car_v1_start).try_into().unwrap(),
+        index_offset.try_into().unwrap(),
+    );
+    writer.write_header(&header).await?;
+
+    let header_v1 = CarV1Header::new(vec![root.unwrap()]);
+    writer.write_v1_header(&header_v1).await?;
+
+    writer.finish().await?;
+
+    Ok(root.unwrap())
+}
+
+async fn balanced_import_unixfs<Src, Out>(
+    mut source: Src,
+    mut output: Out,
+    chunk_size: usize,
+    tree_width: usize,
+) -> Result<Cid, Error>
+where
+    Src: AsyncRead + Unpin,
+    Out: AsyncWrite + AsyncSeek + Unpin,
+{
+    let chunker = async_stream::try_stream! {
```

Comment:
If you're using the same approach as the other balanced import:
1. Reference to the other implementation for details (someone may read this and question why it's done the way it it)
2. Maybe there's a way of having this chunk be the same for both? (no macros)

Reply:
added comment